spec/outputs/dynatrace_spec.rb

```diff
@@ -19,133 +19,97 @@
 require 'logstash/event'
 require 'sinatra'
 require 'insist'
-
-PORT = rand(65_535 - 1024) + 1025
-
-# NOTE: that Sinatra startup and shutdown messages are directly logged to stderr so
-# it is not really possible to disable them without reopening stderr which is not advisable.
-#
-# == Sinatra (v1.4.6) has taken the stage on 51572 for development with backup from WEBrick
-# == Sinatra has ended his set (crowd applauds)
-#
-class TestApp < Sinatra::Base
-  # disable WEBrick logging
-  def self.server_settings
-    { AccessLog: [], Logger: WEBrick::BasicLog.new(nil, WEBrick::BasicLog::FATAL) }
-  end
-
-  class << self
-    attr_accessor :last_request
-
-    def clear
-      self.last_request = nil
-    end
-  end
-
-  post '/good' do
-    self.class.last_request = request
-    [204, '']
-  end
-
-  post '/bad' do
-    self.class.last_request = request
-    [400, 'Bad']
-  end
-end
-
-RSpec.configure do |config|
-  # http://stackoverflow.com/questions/6557079/start-and-call-ruby-http-server-in-the-same-script
-  def sinatra_run_wait(app, opts)
-    queue = Queue.new
-
-    t = java.lang.Thread.new(
-      proc do
-        begin
-          app.run!(opts) do |_server|
-            queue.push('started')
-          end
-        rescue StandardError => e
-          puts "Error in webserver thread #{e}"
-          # ignore
-        end
-      end
-    )
-    t.daemon = true
-    t.start
-    queue.pop # blocks until the run! callback runs
-  end
-
-  config.before(:suite) do
-    sinatra_run_wait(TestApp, port: PORT, server: 'webrick')
-    puts "Test webserver on port #{PORT}"
-  end
-end
+require 'net/http'
+require 'json'
 
 describe LogStash::Outputs::Dynatrace do
-  let(:port) { PORT }
-  let(:event) do
-    LogStash::Event.new({ 'message' => 'hi' })
+  let(:events) do
+    [
+      LogStash::Event.new({ 'message' => 'message 1', '@timestamp' => "2021-06-25T15:46:45.693Z" }),
+      LogStash::Event.new({ 'message' => 'message 2', '@timestamp' => "2021-06-25T15:46:46.693Z" }),
+    ]
   end
-  let(:url) { "http://localhost:#{port}/good" }
+  let(:url) { "http://localhost/good" }
   let(:key) { 'api.key' }
   let(:subject) { LogStash::Outputs::Dynatrace.new({ 'api_key' => key, 'ingest_endpoint_url' => url }) }
 
   before do
     subject.register
-    allow(subject).to receive(:log_failure).with(any_args)
+    subject.plugin_version = "1.2.3"
```

Comment:
Why did this work without it earlier?

Reply:
User agent was not asserted on specifically before